tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/IR/hlo_ops.cc

```diff
@@ -4849,9 +4849,37 @@ static LogicalResult EliminateRedundantTranspse(TransposeOp op,
   return success();
 }
 
+// transpose(broadcast_in_dim(X)) => broadcast_in_dim(X)
+static LogicalResult EliminateBroadcastInDimTranspose(
+    TransposeOp op, PatternRewriter& rewriter) {
+  auto broadcast_in_dim_op = op.operand().getDefiningOp<BroadcastInDimOp>();
+  if (!broadcast_in_dim_op) {
+    return failure();
+  }
+  DenseIntElementsAttr broadcast_dimensions =
+      broadcast_in_dim_op.broadcast_dimensions();
+  DenseIntElementsAttr permutation = op.permutation();
+  SmallVector<int64_t> new_broadcast_dimensions;
+  for (APInt dimension : broadcast_dimensions) {
+    int64_t index = 0;
+    for (APInt p : permutation) {
```

Comment:
```suggestion
  for (auto dimension : broadcast_dimensions.getValues<int64_t>()) {
    int64_t index = 0;
    for (autp p : permutation.getValues<int64_t>()) {
```

And that's a good example of why `auto` is obfuscating the code :)